File: app/src/main/java/bb/incognito/injection/component/ApplicationComponent.java
package bb.incognito.injection.component;

import android.app.Application;
import javax.inject.Singleton;
import bb.incognito.data.DataManager;
import bb.incognito.injection.module.ApplicationModule;
import bb.incognito.view.MainActivity;
import dagger.Component;

@Singleton
@Component(modules = ApplicationModule.class)
public interface ApplicationComponent {

    void inject(MainActivity mainActivity);

    Application application();
    DataManager dataManager();
}

File: app/src/main/java/com/koktajlbar/incognitobook/components/ApplicationComponent.kt
package com.koktajlbar.incognitobook.components

import android.content.Context
import com.koktajlbar.incognitobook.IncognitoBook
import com.koktajlbar.incognitobook.modules.ApplicationModule
import com.koktajlbar.incognitobook.modules.CocktailsListModule
import com.koktajlbar.incognitobook.modules.NetworkModule
import dagger.BindsInstance
import dagger.Component
import dagger.android.AndroidInjector
import dagger.android.support.AndroidSupportInjectionModule
import javax.inject.Singleton

@Singleton
@Component(
        modules = [
            ApplicationModule::class,
            NetworkModule::class,
            AndroidSupportInjectionModule::class,
            CocktailsListModule::class
        ]
)
interface ApplicationComponent : AndroidInjector<IncognitoBook> {
    @Component.Factory
    interface Factory {
        fun create(@BindsInstance applicationContext: Context): ApplicationComponent
    }
}